libusb/core.c

```diff
@@ -2202,10 +2202,6 @@ int API_EXPORTED libusb_set_option(libusb_context *ctx,
 		return r;
 	}
 
-	if (option >= LIBUSB_OPTION_MAX) {
```

Comment:
This removal changes behaviour though... if you don't get into `if (NULL == ctx)` then `return LIBUSB_ERROR_INVALID_PARAM` no longer happens.

If you remove `case LIBUSB_OPTION_MAX:` but leave the `default` would that shut it up?

Reply:
That's what I did in commit cd8078d but now we have a GCC warning instead...

Reply:
Exactly. You either get complaints about not covering all enum values in the switch, or you have dead code.

The only behaviour change is that the context is resolved though. I do think that this is reasonable, in particular for something that is obviously a programming/dependency version error. Like, it would almost be reasonable to just assert() here.

Reply:
`option >= LIBUSB_OPTION_MAX` is possible if some old libusb dynamic library is installed, and some new application uses some new future enum value. You'd want to `return LIBUSB_ERROR_INVALID_PARAM` in that case I think, which in master is done both by the early check and the later switch statement's default case.

If you remove the early check, as in this commit, then like I said there is a behaviour change (some additional code is run), but I'm not sure if it matters or not.